cirq-core/cirq/protocols/pauli_expansion_protocol.py

```diff
@@ -86,6 +86,10 @@ def pauli_expansion(
         return default
 
     num_qubits = matrix.shape[0].bit_length() - 1
+
+    if num_qubits == 0:
+        return default
```

Comment:
Why is this needed? Please add explicit tests.

Reply:
@viathor Need some numpy foo. I tracked this down to a root cause, and it appears `hilbert_schmidt_inner_product` fails for 1x1 matrices.

The reason tests now fail otherwise (without the above workaround), whereas they passed before, is that PauliStringPhasor did not define a _pauli_expansion_ method. But now since it inherits GateOperation, it does. There's a consistent_protocols test that checks this for an empty pauli string. This was fine before because the expansion protocol was not implemented, but now it is implemented and gets checked. The unitary matrix it uses in this case is 1x1, which fails at the inner_product step.

Reply:
@viathor nvm, the problem is actually in `operator_spaces.kron_bases`, where the initial array is missing a dimension. That's an easy fix. I'll fix that and add a test.